### PR TITLE
config: split config up

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
config

### Description
This PR splits the config up. This is necessary to correctly work with nestjs.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

File: src/media/media.service.spec.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ describe('MediaService', () => {
       imports: [
         ConfigModule.forRoot({
           isGlobal: true,
-          load: [appConfigMock],
+          load: [appConfigMock, mediaConfigMock],
```
Comment (davidmehren) [inline src/media/media.service.spec.ts:43]: `appConfigMock` can be removed

File: test/public-api/media.e2e-spec.ts [hunk 2]
```diff
@@ -31,7 +32,7 @@ describe('Notes', () => {
       imports: [
         ConfigModule.forRoot({
           isGlobal: true,
-          load: [appConfigMock],
+          load: [appConfigMock, mediaConfigMock],
```
Comment (davidmehren) [inline test/public-api/media.e2e-spec.ts:35]: `appConfigMock` can be removed

File: test/public-api/notes.e2e-spec.ts [hunk 3]
```diff
@@ -27,7 +28,7 @@ describe('Notes', () => {
       imports: [
         ConfigModule.forRoot({
           isGlobal: true,
-          load: [appConfigMock],
+          load: [appConfigMock, mediaConfigMock],
```
Comment (davidmehren) [inline test/public-api/notes.e2e-spec.ts:31]: `appConfigMock` can be removed

**Review by davidmehren — CHANGES_REQUESTED** [review]
In addition to the bits I commented on, this PR exacerbates an existing problem:  
The error messages when the validation of the config fails are unhelpful. Previously, they didn't reference the name of the environment variable, but you had an idea how that could look like when the error complained about `database.username` not being present.

Now, the error will only say that `username` is missing, which is completely not helpful.
We need to improve the messages here ASAP.

**Review by davidmehren — APPROVED** [review]

**Pull request merged**